Sdk/Speckle.Connectors.Common/Operations/ReceiveOperation.cs

```diff
@@ -50,15 +49,12 @@ CancellationToken cancellationToken
       .ConfigureAwait(false);
 
     cancellationToken.ThrowIfCancellationRequested();
-    await apiClient.Version.Received(
-      new(version.id, receiveInfo.ProjectId, receiveInfo.SourceApplication),
-      cancellationToken
-    );
+    await receiveVersionRetriever.VersionReceived(account, version, receiveInfo, cancellationToken);
```

Comment:
Having yet another layer of abstraction around client functions is not ideal....
Is it possible to to moq a `client` or the `VersionResource`...

There's enough functions that require the `Client` that I think its worth us ensuring we create a nice pattern around unit testing it. Especially seeing as we already interface out the `IClientFactory`

Reply:
actually this is still going to be necessary because the client object gives a version resource which can't be mocked